include/deal.II/grid/grid_tools.h

```diff
@@ -1285,6 +1285,269 @@ namespace GridTools
       const bool                                             perform_handshake,
       const bool enforce_unique_mapping = false);
 
+
+    /**
+     * A function that fills DistributedComputeIntersectionLocationsInternal.
+     * @p intersection_requests are vertices of cells which describe the
+     * entities we want to search intersections for.
+     * The template parameter @p request_dim provides the dimension of the
+     * requesting intersection, e.g., if the intersection request corresponds
+     * to faces request_dim = dim-1 and for cells request_dim = dim.
+     */
+    template <int request_dim,
+              int dim,
+              int spacedim,
+              typename IntersectionRequest>
+    DistributedComputeIntersectionLocationsInternal<dim,
+                                                    spacedim,
+                                                    IntersectionRequest>
+    distributed_compute_intersection_locations(
+      const GridTools::Cache<dim, spacedim> & cache,
+      const std::vector<IntersectionRequest> &intersection_requests,
+      const std::vector<std::vector<BoundingBox<spacedim>>> &global_bboxes,
+      const std::vector<bool> &                              marked_vertices,
+      const double                                           tolerance)
+    {
+      using IntersectionAnswer =
+        typename DistributedComputeIntersectionLocationsInternal<
+          dim,
+          spacedim,
+          IntersectionRequest>::IntersectionAnswer;
+      const auto comm = cache.get_triangulation().get_communicator();
+
+      DistributedComputeIntersectionLocationsInternal<dim,
+                                                      spacedim,
+                                                      IntersectionRequest>
+        result;
+
+      auto &send_components = result.send_components;
+      auto &recv_components = result.recv_components;
+      auto &recv_ptrs       = result.recv_ptrs;
+      auto &send_ranks      = result.send_ranks;
+      auto &recv_ranks      = result.recv_ranks;
+
+      // construct entities we use to search for potential owners
+      std::vector<dealii::BoundingBox<dim>> request_bboxes;
+      request_bboxes.reserve(intersection_requests.size());
+      for (const auto &intersection_request : intersection_requests)
+        request_bboxes.emplace_back(
+          dealii::BoundingBox<dim>(intersection_request));
```

Comment:
Is this needed? I would have hoped that it is done automatically in `guess_owners_of_entities`!?

Reply:
I think this should work :+1:

Reply:
Done in https://github.com/dealii/dealii/pull/15165